File: cpp/questions/groupAnagrams.cpp
#include <vector>
#include <string>
#include <unordered_map>
#include <algorithm>

using namespace std;

class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        std::unordered_map<std::string, std::vector<std::string>> hm;
        for (auto& s: strs) {
            std::string k = s;
            std::sort(k.begin(), k.end());
            hm[k].push_back(s);
        }
        std::vector<std::vector<std::string>> ans;
        for (const auto& p: hm) {
            ans.push_back(std::move(p.second));
        }
        return ans;
    }
};

File: rust/questions/group_anagrams.rs
use std::collections::HashMap;

impl Solution {
    pub fn group_anagrams(strs: Vec<String>) -> Vec<Vec<String>> {
        let mut hm: HashMap<Vec<u8>, Vec<String>> = HashMap::new();
        for s in strs {
            let mut sv = s.as_bytes().to_vec();
            sv.sort_unstable();
            hm.entry(sv).or_default().push(s);
        }
        hm.into_values().collect()
    }
}
